src/utilities/map.ts

```diff
@@ -34,3 +34,30 @@ export function getTileElement(x: number, y: number, elementIdx: number): TileEl
 	const tile = getTileByCoords(x, y);
 	return tile.getElement(elementIdx);
 }
+
+/**
+ * Finds the track element index on a tile that matches the given track location (car.trackLocation).
+ */
+export function findTrackElementIndex(trackLocation: CarTrackLocation): number | null 
+{
+	const trackTile = map.getTile(toTileUnit(trackLocation.x), toTileUnit(trackLocation.y));
+	const elements = trackTile.elements;
+	for (let i = 0; i < elements.length; i++) 
+	{
+		const element = elements[i];
+		if (element.type === "track") 
+		{
+			const trackIterator = map.getTrackIterator(trackLocation, i);
```

Comment:
Do we really need a track iterator for this check? 🤔

Reply:
well it's mostly the z field that's important to figure out. We can't figure the tile `i` for the track location we read from the car so we need to iterate the tile until we find a matching element at the right z, and it appears to only be available via the iterator.